docs/src/rules/no-useless-assignment.md

```diff
@@ -3,11 +3,25 @@ title: no-useless-assignment
 rule_type: suggestion
 related_rules:
 - no-unused-vars
+further_reading:
+- https://en.wikipedia.org/wiki/Dead_store
+- https://www.npmjs.com/package/eslint-plugin-no-useless-assign
+- https://sonarsource.atlassian.net/browse/RSPEC-1854
+- https://rules.sonarsource.com/javascript/RSPEC-1854/
+- https://rules.sonarsource.com/typescript/RSPEC-1854/
```

Comment:
The previous three lines all basically point to the same information. I'd suggest we keep just the second one, which is JavaScript-specific.